components/ChallengeMaterial.tsx

```diff
@@ -1,16 +1,20 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { Challenge } from '../@types/challenge'
 
+type CurrentChallengeID = string | null
+
 type ChallengeTitleCardProps = {
+  key: string
+  id: string
   title: string
   challengeNum: number
   currentState?: string
   active?: boolean
+  setCurrentChallenge: React.Dispatch<CurrentChallengeID>
 }
 
 type ChallengeQuestionCardProps = {
-  title: string
-  question: string
+  currentChallenge: Challenge | undefined
```

Comment:
Why `Challenge | undefined`?

Reply:
ill update this to assume that currentChallenge will be the first challenge unless there is a more recent active one.